integration-tests/tests/aws/nino-check/nino-check-unhappy.test.ts

```diff
@@ -68,151 +77,149 @@ describe("nino-check-unhappy", () => {
-    await clearItems(personIdentityTableName, {
-      sessionId: input.sessionId,
-    });
-    const startExecutionResult = await executeStepFunction(
-      input,
-      output.NinoCheckStateMachineArn
-    );
-    expect(startExecutionResult.output).toBe(
-      '{"error":"No user found for given nino"}'
-    );
-  });
+  const startExecutionResult = await executeStepFunction(
+    input,
+    output.NinoCheckStateMachineArn
+  );
 
-  it("should fail when session id is invalid", async () => {
-    await clearItems(sessionTableName, {
-      sessionId: input.sessionId,
-    });
-    const startExecutionResult = await executeStepFunction(
-      input,
-      output.NinoCheckStateMachineArn
-    );
-    expect(startExecutionResult.output).toBe(
-      '{"error":"Session is not valid or has expired"}'
-    );
-    expect(startExecutionResult.status).toBe("SUCCEEDED");
+  expect(startExecutionResult.output).toBe(
+    '{"error":"Maximum number of attempts exceeded"}'
+  );
+});
+
+it("should fail when there is no user present for given nino", async () => {
+  await clearItems(personIdentityTableName, {
+    sessionId: input.sessionId,
   });
 
-  it("should fail when user record already present in nino user table", async () => {
-    await populateTable(
-      {
-        sessionId: "123456789",
-        nino: "AA000003D",
-      },
-      output.NinoUsersTable
-    );
-    const startExecutionResult = await executeStepFunction(
-      {
-        sessionId: "123456789",
-        nino: "AA000003D",
-      },
-      output.NinoCheckStateMachineArn
-    );
+  const startExecutionResult = await executeStepFunction(
+    input,
+    output.NinoCheckStateMachineArn
+  );
 
-    expect(startExecutionResult.status).toBe("FAILED");
+  expect(startExecutionResult.output).toBe(
+    '{"error":"No user found for given nino"}'
+  );
+});
+
+it("should fail when session id is invalid", async () => {
+  await clearItems(sessionTableName, {
+    sessionId: input.sessionId,
   });
 
-  it("should fail when user nino does not match with HMRC DB", async () => {
-    const startExecutionResult = await executeStepFunction(
-      input,
-      output.NinoCheckStateMachineArn
-    );
-    expect(startExecutionResult.output).toBe(
-      '{"error":"CID returned no record"}'
-    );
+  const startExecutionResult = await executeStepFunction(
+    input,
+    output.NinoCheckStateMachineArn
+  );
+
+  expect(startExecutionResult.output).toBe(
+    '{"error":"Session is not valid or has expired"}'
+  );
+
+  expect(startExecutionResult.status).toBe("SUCCEEDED");
+});
+
+it("should fail when user record already present in nino user table", async () => {
+  await populateTable(output.NinoUsersTable as string, {
+    sessionId: "123456789",
+    nino: "AA000003D",
   });
 
-  it("should throw an error when url is unavailable", async () => {
-    const urlParameterName = `/${process.env.STACK_NAME}/NinoCheckUrl`;
+  const startExecutionResult = await executeStepFunction(
```

Comment:
Can we do similar logic here that we did with the bearerToken? I can see this block failing for the same reasons. 

So remove: 

```
const currentURL = (
    await getSSMParamter({
      Name: urlParameterName,
    })
  ).Parameter?.Value as string;
```
and move this to the before block:

```
await ssmParamterUpdate({
    Name: urlParameterName,
    Value: currentURL,
    Type: "String",
    Overwrite: true,
  });
```

Reply:
We probably will want to store the "current" URL so query once before the tests run